app/Http/Controllers/ReadingItemController.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\ReadingItem;
+use Illuminate\Http\Request;
+
+class ReadingItemController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return view('reading-items.index')->with([
+            'reading_items' => ReadingItem::all()->sortByDesc('created_at'),
+        ]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        ReadingItem::create([
+            'topic' => $request->input('topic'),
+            'url' => $request->input('url'),
+            'recommended_by' => $request->input('recommended_by'),
+        ]);
+        return json_encode(['success' => true]);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
```

Comment:
Would it be a good idea to remove all this comment blocks. we don't need them. This comment is anyways violating DRY and representing incorrect information. The method doesn't return `\Illuminate\Http\Response` today.

Reply:
I agree that it doesn't return `\Illuminate\Http\Response` today. Just curious if someone uses this method later, having a return type specified may help developers submit PRs with the proper return type.
In that case, should we remove the method as well and put it back whenever required with proper doc comments?

Moreover, on violating DRY, IMO it's correct until the doc block doesn't specify anything extra that we can't get out of the function signature. For a resource controller, I think, having the description of `create` and `store` methods can help a beginner to differentiate between these similar names or maybe for `edit` and `update`.

In the end, it may be a debatable topic to have doc-blocks, but I think they aren't for us, but for the developers coming after us. I found this article helpful:
https://www.brandonsavage.net/documenting-your-code-in-a-useful-way/

Reply:
Having a return type specified may help when the method is implemented. However, just considering the realities of human constraints, In this [PR](https://github.com/ColoredCow/weeklydose-service/pull/1/commits/3bd2e719851499a806d333b5f3abf5f20c605f53) you missed returning an objects of `\Illuminate\Http\Response` (store method of `ReadingItemController`).

I think it's good to mention, if you know in advance, that an object of a particular type should be sent back. However, it's important to follow it as well and if you don't then change the DocBlock with the appropriate message, which was missed in this case. Again, it's opinion, I leave the final call to you.

Reply:
Totally agree with the miss. When returning the JSON object, the DocBlock should've been updated accordingly.

I've updated the return types of the methods. It needs to be added to the practice while making changes to update the return type. Please take a look and let me know.